modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/construction/ConstructionService.scala

```diff
@@ -39,5 +41,12 @@ object ConstructionService {
       .map(TransactionIdentifier(_))
       .attemptT
       .leftMap(_ => MalformedTransaction)
+
+    def getAccountIdentifiers(operations: List[Operation]): Option[List[AccountIdentifier]] = {
+      val negativeValueOperations = operations.filter(_.amount.exists(_.value < 0))
+
+      val test = Some(negativeValueOperations.flatMap(_.account)).filter(test => test.nonEmpty)
+      test
+    }
```

Comment:
we've already rewritten that on slack, please take a look there

Reply:
That code depended on `amount` being an `Option[Either[NegAmount, PosAmount]]`. It is now just a `Long Refined Not[Equal[0L]]`. I'll play around with this to see if I can make something cleaner.

Reply:
This is what I came up with

```
def getAccountIdentifiers(operations: List[Operation]): Option[NonEmptyList[AccountIdentifier]] = {
      val accountIdentifiers =
        operations.filter(_.amount.exists(_.value.value.value < 0)).flatMap(_.account)

      Option.when(accountIdentifiers.nonEmpty)(NonEmptyList.fromList(accountIdentifiers)).flatten
    }
```